lib/config/options.rb

```diff
@@ -201,15 +201,19 @@ def __convert(h) #:nodoc:
 
       h.each do |k, v|
         k = k.to_s if !k.respond_to?(:to_sym) && k.respond_to?(:to_s)
-        s.new_ostruct_member(k)
 
         if v.is_a?(Hash)
           v = v["type"] == "hash" ? v["contents"] : __convert(v)
         elsif v.is_a?(Array)
           v = v.collect { |e| e.instance_of?(Hash) ? __convert(e) : e }
         end
 
-        s.send("#{k}=".to_sym, v)
+        if s.respond_to?(:[]=)
```

Comment:
Can you add a comment about why this conditional is needed?

Reply:
This was meant for backward compatibility with older Ruby versions. Now I noticed that the version `#[]=` was introduced is way older than the version `config` claims to support... https://github.com/ruby/ruby/commit/e44e356b53828d6468114b30e5c169296896f9b1#diff-f0ac5119f74bfda273347c9925e910d7

Therefore I'm going to remove this conditional, rather than adding comments.